### PR TITLE
Proposed solution to ECAL emulator PR issue 

### PR DESCRIPTION
**Pull request by atishelmanch**
#### PR description:

The purpose of this pull request is to address issue https://github.com/cms-sw/cmssw/issues/33425, raised after merging the ECAL emulator update PR https://github.com/cms-sw/cmssw/pull/33349. This PR proposes two solutions to the issue:

1) Non-thread-safe cout statements are replaced with LogVerbatim or LogInfo. 
2) Previously caught exceptions are now rethrown as cms exceptions. 

#### PR validation:

- All runTheMatrix tests have passed 
- Code checks and code formatting applied in commits 
- Tested each step of the way and successfully ran with an ECAL trigger cmssw configuration file [1] 

cc: @thomreis @valsdav 

[1] https://github.com/CMS-ECAL-Trigger-Group/ETTAnalyzer

### CODE REVIEW

**Comment by tvami** [comment]
@cmsbuild , please test

File: CondTools/Ecal/src/EcalTPGOddWeightIdMapHandler.cc [hunk 1]
```diff
@@ -213,44 +212,44 @@ void popcon::EcalTPGOddWeightIdMapHandler::getNewObjects() {
...
 
           } catch (std::exception& e) {
-            std::cout << "ERROR: THIS CONFIG DOES NOT EXIST: tag=" << the_config_tag
-                      << " version=" << the_config_version << std::endl;
-            std::cout << e.what() << std::endl;
+            throw cms::Exception("FileReadError") << "ERROR: THIS CONFIG DOES NOT EXIST: tag=" << the_config_tag
+                                                  << " version=" << the_config_version << "\n"
+                                                  << e.what();
             m_i_run_number = irun;
           }
```
Comment (tvami) [inline CondTools/Ecal/src/EcalTPGOddWeightIdMapHandler.cc:224]: I think you should not catch the exception, but let the framework do it, please see 7.11 in https://cms-sw.github.io/cms_coding_rules.html

**Review by makortel — COMMENTED** [review]
In general catching and rethrowing exceptions is ok as long as there is good reason for it (like adding more relevant context information or exception being the only or cheapest way of checking errors of external code). Any code interacting directly with DB is somewhat special case anyway.

I did not see anything particularly worrisome in the changes of this PR.

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
+alca
 * there are legacy module issues, for another PR, I created a reminder here: https://github.com/cms-AlCaDB/AlCaTools/issues/41
 * SA finds other warnings too in the same package (Dead assignment, Memory leak, Result of operation is garbage or undefined, non-const static variable) -- this could be fixed in a follow-up as well

**Comment by perrotta** [comment]
+1
- As already mentioned in https://github.com/cms-sw/cmssw/pull/35557#issuecomment-939301559. it would be wise to solve all the (many) issues reported by the Static Analizer. Ok for having it done in a different PR

**Pull request merged**